src/regex.cr

```diff
@@ -526,7 +526,7 @@ class Regex
   # re.match("Skiing")   # => Regex::MatchData("Skiing")
   # re.match("sledding") # => Regex::MatchData("sledding")
   # ```
-  def +(other) : Regex
+  def +(other : Regex) : Regex
```

Comment:
That's what `Regex.union` accepts:

```suggestion
  def +(other : Regex | String) : Regex
```

Reply:
It seems a bit odd though. I don't think the `String` overload is much useful. Maybe we should deprecate that?

For this PR it might just be best to remove the type restriction. We can follow up with a dedicated change.